.github/workflows/publish.yaml

```diff
@@ -0,0 +1,14 @@
+# A CI configuration to auto-publish pub packages.
+
+name: Publish
+
+on:
+  pull_request:
+    branches: [ master ]
+  push:
+    tags: [ 'v[0-9]+.[0-9]+.[0-9]+*' ]
```

Comment:
(I think this is a glorified glob pattern, not a regexp. Very confusing to read :thinking:.)

The pattern can drop the last `+`. It's probably benign, I don't think a glob pattern can get catastrophic backtracking, especially when there is nothing after the `*`. With any luck, the implementation will go to the `*` after seeing the first digit.

Reply:
The pattern was intended to be able to capture `x.y.x-something`, for completeness. But thinking about it, if we really only ever intend to support publishing release versions via this automation, than dropping the suffix makes sense.